### ISSUE TITLE
Skip all release automation related tests on platforms which are not Linux

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4154]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4154|
|Type|task|
|Created|2009-12-06 20:54:38Z|
|Branch|https://github.com/twisted/twisted/tree/skip-release-tests-4154|

Since the release is done on Linux, and these release tools are **explicitly** for our use only, it's a bit of a waste to run the tests for them on other platforms, and an even bigger waste to spend development effort making the code portable and so on.

So, let's skip the tests everywhere but on Linux.

<details><summary>Searchable metadata</summary>

```
trac-id__4154 4154
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_skip_release_tests_4154 branches/skip-release-tests-4154
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__tests tests
time__1260132878000000 1260132878000000
changetime__1261148190000000 1261148190000000
version__None None
owner__ 
cc__radix cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [27687]) Branching to 'skip-release-tests-4154'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

I'm assuming they should be skipped on OSX as well?

I guess the following tests are not part of the release process? Seems to me they can be skipped though, but I could be wrong.

```
twisted.python.test.test_dist
  GetScriptsTest
    test_getScriptsTopLevel ...                                            [OK]
    test_noScriptsInSVN ...                                                [OK]
    test_noScriptsInSubproject ...                                         [OK]
    test_scriptsInRelease ...                                              [OK]
    test_scriptsInSVN ...                                                  [OK]
  GetVersionTest
    test_getVersionCore ...                                                [OK]
    test_getVersionOther ...                                               [OK]
  SetupTest
    test_conditionalExtensions ...                                         [OK]
    test_win32Definition ...                                               [OK]
```

And why are the ```BuildAllTarballsTest``` and ```DistributionBuilderTest``` tests not being skipped?

```
twisted.python.test.test_release
  APIBuilderTestCase
    test_apiBuilderScriptMain ...                                     [SKIPPED]
    test_apiBuilderScriptMainRequiresTwoArguments ...                 [SKIPPED]
    test_build ...                                                    [SKIPPED]
    test_buildWithPolicy ...                                          [SKIPPED]
  BookBuilderTests
    test_build ...                                                    [SKIPPED]
    test_buildPDF ...                                                 [SKIPPED]
    test_buildPDFLongPath ...                                         [SKIPPED]
    test_buildPDFRejectsInvalidBookFilename ...                       [SKIPPED]
    test_buildPDFRunsLaTeXThreeTimes ...                              [SKIPPED]
    test_buildRemovesTemporaryLaTeXFiles ...                          [SKIPPED]
    test_buildTeX ...                                                 [SKIPPED]
    test_buildTeXOnlyBuildsXHTML ...                                  [SKIPPED]
    test_buildTeXRejectsInvalidDirectory ...                          [SKIPPED]
    test_failedCommandProvidesOutput ...                              [SKIPPED]
    test_noSideEffects ...                                            [SKIPPED]
    test_runFailed ...                                                [SKIPPED]
    test_runSignaled ...                                              [SKIPPED]
    test_runSuccess ...                                               [SKIPPED]
    test_stdout ...                                                   [SKIPPED]
  BuildAllTarballsTest
    test_buildAllTarballs ...                                              [OK]
    test_buildAllTarballsEnsuresCleanCheckout ...                          [OK]
    test_buildAllTarballsEnsuresExistingCheckout ...                       [OK]
  ChangeVersionTest
    test_changeAllProjectVersions ...                                 [SKIPPED]
    test_changeVersionInFile ...                                      [SKIPPED]
    test_getNextVersion ...                                           [SKIPPED]
    test_getNextVersionAfterYearChange ...                            [SKIPPED]
  DistributionBuilderTest
    test_apiBaseURL ...                                                    [OK]
    test_coreProjectLayout ...                                             [OK]
    test_minimalSubProjectLayout ...                                       [OK]
    test_subProjectDocBuilding ...                                         [OK]
    test_subProjectLayout ...                                              [OK]
    test_twistedDistribution ...                                           [OK]
    test_twistedDistributionExcludesWeb2AndVFSAndAdmin ...                 [OK]
  DocBuilderTestCase
    test_apiLinking ...                                                    [OK]
    test_build ...                                                         [OK]
    test_deleteInput ...                                                   [OK]
    test_doNotDeleteInput ...                                              [OK]
    test_getLinkrelToParentDirectory ...                                   [OK]
    test_getLinkrelToSameDirectory ...                                     [OK]
    test_getLinkrelToSibling ...                                           [OK]
    test_getLinkrelToUncle ...                                             [OK]
    test_noDocumentsFound ...                                              [OK]
    test_parentDocumentLinking ...                                         [OK]
    test_siblingDirectoryDocumentLinking ...                               [OK]
  FilePathDeltaTest
    test_filePathDeltaSiblingDir ...                                  [SKIPPED]
    test_filePathDeltaSubdir ...                                      [SKIPPED]
    test_filePathNoCommonElements ...                                 [SKIPPED]
  ManBuilderTestCase
    test_build ...                                                         [OK]
    test_noDocumentsFound ...                                              [OK]
    test_toHTML ...                                                        [OK]
  NewsBuilderTests
    test_build ...                                                    [SKIPPED]
    test_buildAll ...                                                 [SKIPPED]
    test_duplicatesMerged ...                                         [SKIPPED]
    test_emptySectionsOmitted ...                                     [SKIPPED]
    test_findBugfixes ...                                             [SKIPPED]
    test_findFeatures ...                                             [SKIPPED]
    test_findMiscellaneous ...                                        [SKIPPED]
    test_findRemovals ...                                             [SKIPPED]
    test_preserveTicketHint ...                                       [SKIPPED]
    test_today ...                                                    [SKIPPED]
    test_writeHeader ...                                              [SKIPPED]
    test_writeMisc ...                                                [SKIPPED]
    test_writeSection ...                                             [SKIPPED]
  ProjectTest
    test_findTwistedStyleProjects ...                                 [SKIPPED]
    test_getVersion ...                                               [SKIPPED]
    test_repr ...                                                     [SKIPPED]
    test_updateTwistedVersionInformation ...                          [SKIPPED]
    test_updateVersion ...                                            [SKIPPED]
  ScriptTests
    test_badNumberOfArgumentsToBuildNews ...                          [SKIPPED]
    test_badNumberOfArgumentsToBuildTarballs ...                      [SKIPPED]
    test_badNumberOfArgumentsToChangeVersionsScript ...               [SKIPPED]
    test_buildNews ...                                                [SKIPPED]
    test_buildTarballsScript ...                                      [SKIPPED]
    test_changeVersions ...                                           [SKIPPED]
    test_changeVersionsWithPrerelease ...                             [SKIPPED]
    test_defaultBuildTarballsScriptBuilder ...                        [SKIPPED]
    test_defaultChangeVersionsVersionChanger ...                      [SKIPPED]
    test_nonIntPartsToChangeVersionsScript ...                        [SKIPPED]
    test_tooManyDotsToChangeVersionsScript ...                        [SKIPPED]
  UtilityTest
    test_chdir ...                                                    [SKIPPED]
    test_replaceInFile ...                                            [SKIPPED]
  VersionWritingTest
    test_replaceProjectVersion ...                                    [SKIPPED]
    test_replaceProjectVersionWithPrerelease ...                      [SKIPPED]
```

Tested on OSX 10.5.8 with Python 2.6.4.

Perhaps the docstring of the module should have a note explaining why the tests are (not) being skipped (on linux)?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [27788]) Update module docstrings; fix the skip logic to make sure it really skips on bad platforms

refs [#4154](https://github.com/twisted/twisted/issues/4154)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

Thanks for the review.

I fixed the skips.  It seems that having skip set to `None` on a test case overrides having a non-None `skip` at the module level.

`twisted.python.dist` is used by `setup.py` so it does need to continue to work on all supported platforms.

I also updated the module docstrings to talk about the Linux-only restriction.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

Cool, +1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [27823]) Merge skip-release-tests-4154

Author: exarkun
Reviewer: thijs, therve
Fixes: [#4154](https://github.com/twisted/twisted/issues/4154)

Skip all of the automated release process unit tests on all platforms except Linux (the
only platform where the automated release process needs to work).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
